src/main/java/de/embl/cba/bdv/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -132,7 +147,29 @@ private void fetchSequenceDescriptionAndViewRegistrations()
 			for ( int setupId = 0; setupId < numSetups; setupId++ )
 			{
 				ViewSetup viewSetup = createViewSetup( setupId );
-				int setupTimepoints = ( int ) setupToAttributes.get( setupId ).getDimensions()[ T ];
+				int setupTimepoints = 1;
+				is5D = false;
+				if (setupToAttributes.get(setupId).getNumDimensions() > 4 && axesMap.containsKey("t")) {
+					setupTimepoints = (int) setupToAttributes.get(setupId).getDimensions()[T];
+					is5D = true;
+				} else if (setupToAttributes.get(setupId).getNumDimensions() > 4) {
+					setupTimepoints = (int) setupToAttributes.get(setupId).getDimensions()[T];
+					is5D = true;
+				}
+
+				if (setupToAttributes.get(setupId).getNumDimensions() == 4 && axesMap.containsKey("t") && !axesMap.containsKey("c")) {
```

Comment:
This coundition and the next one (line 164) are identical in this context and it should be enough to check for 
`nDim == 4 && axesMap.containsKey("t")`